tests/validation/tests/v3_api/test_hpa.py

```diff
@@ -0,0 +1,533 @@
+import pytest
+from rancher import ApiError
+from .common import *  # NOQA
+
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None}
+
+
+def test_create_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa, workload = create_hpa(p_client, ns)
+    p_client.delete(hpa, workload)
+
+
+def test_edit_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa, workload = edit_hpa(p_client, ns)
+    p_client.delete(hpa, workload)
+
+
+def test_delete_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa, workload = create_hpa(p_client, ns)
+    delete_hpa(p_client, hpa, ns)
+    p_client.delete(workload)
```

Comment:
Can you include cases relating to deleting hpa that are still associated with workloads ?

Reply:
As discussed, its not required